doc/policy_tutorial/POLICY_SCAFFOLD.md

```diff
@@ -0,0 +1,94 @@
+```
+
+As you can see from the output of the generate policy command a few files have been created. These artifacts related to our policy are located in three different directories:
+
+* t/ – this directory contains all Nginx integration tests
+* src/gateway/apicast/policy – this directory contains the source code and configuration schemas of all policies. Our policy resides in the subdirectory of hello_world
+* spec/policy – this directory contains the unit tests of all policies. The unit tests for our policy resides in the subdirectory of hello_world
+
+So the policy scaffolding utility not only generates a scaffold for our policy, but also the files for a configuration schema, unit tests and integration tests. Let’s have a look at these files.
+
+The source code of our policy residing in the directory src/gateway/apicast/policy/hello_world contains three files.
+
+* init.lua - all policies contain this init.lua file. It contains 1 line importing (require in Lua) our policy. It should not be modified.
+* aplicast-policy.json - The APIcast gateway is configured using a json document. Policies requiring configuration also use this json document. The apicast-policy.json file is a json schema file were configuration properties for the policy can be defined. We will look into configuration properties and this file in more detail in our next part of the workshop.
+```json
+{
+  "$schema": "http://apicast.io/policy-v1/schema#manifest#",
+  "name": "hello_world",
+  "summary": "TODO: write policy summary",
+  "description": [
+      "TODO: Write policy description"
+  ],
+  "version": "builtin",
+  "configuration": {
+    "type": "object",
+    "properties": { }
+  }
+}
+```
+* hello_world.lua - This is the actual source code of our policy, which at the moment does not contain much.
+```lua
+-- This is a hello_world description.
+local policy = require('apicast.policy')
+local _M = policy.new('hello_world')
+local new = _M.new
+--- Initialize a hello_world
+-- @tparam[opt] table config Policy configuration.
+function _M.new(config)
+  local self = new(config)
+  return self
+end
+return _M
+```
+
+The first two lines import the APIcast policy module an instantiate a new policy with hello_world as an argument. This returns a module itself which is implemented using a Lua table. Lua is not an Object Oriented language from itself but tables (and especially metatables) can mimic objects. The third line stores a reference to a function new which is defined below. The new function takes a config variable as argument, but as of now nothing is done with is. The new method simply returns itself. Finally the module representing our policy is returned. This is done so other components importing this policy module retrieve the table and can invoke all functions and variables stored in the policy.
+We won’t cover all the files in details here since we are going to touch these in upcoming series when we flesh out our policy with functionality.
```

Comment:
All the files are not covered in detail here since we are going to touch these in an upcoming series when we flesh out our policy with functionality.